repose-aggregator/commons/utilities/src/main/java/org/openrepose/commons/utils/logging/apache/format/stock/TimeReceivedHandler.java

```diff
@@ -28,8 +29,20 @@
 
 public class TimeReceivedHandler implements FormatterLogic {
 
+    private final static String DEFAULT_DATE_FORMAT = "dd-MM-yyyy-HH:mm:ss.SSS";
+
+    private final String dateFormat;
+
+    public TimeReceivedHandler(final String dateFormat) {
+        this.dateFormat = StringUtils.isEmpty(dateFormat) ? DEFAULT_DATE_FORMAT : dateFormat;
+    }
+
     @Override
     public String handle(HttpServletRequest request, HttpServletResponse response) {
-        return new SimpleDateFormat("dd-MM-yyyy-HH:mm:ss.SSS").format(Calendar.getInstance().getTime());
+        return new SimpleDateFormat(dateFormat).format(Calendar.getInstance().getTime());
+    }
+
+    public String getDateFormat() {
```

Comment:
HttpLogFormatterTest.java has some string parsing tests, and unfortunately it's forced to be more like an integration test than a unit test.  The existing tests just throw some data in and then verify it gets the correct data out, but the TimeReceivedHandler code is grabbing the current time instead of using some input data I can manipulate.  To verify the parsing code is working, I can either use this method to compare the the date format strings or I could grab the output of the class and use a regex to see if it looks like the correct date format was used.

Should I remove the method and go with the regex check?